connector/connect/common/src/main/scala/org/apache/spark/sql/connect/client/ArtifactManager.scala

```diff
@@ -125,7 +125,7 @@ class ArtifactManager(
       .addAllNames(Arrays.asList(artifactName))
       .build()
     val response = bstub.artifactStatus(request)
-    if (response.getSessionId != sessionId) {
+    if (StringUtils.isNotEmpty(response.getSessionId) && response.getSessionId != sessionId) {
```

Comment:
Maybe add a comment that explains why an empty session ID is treated differently. Here and below.

Reply:
added